### PR TITLE
Fix get pets news by id roles

### PR DESCRIPTION
**Pull request by Jakub-Izbicki**

### CODE REVIEW

File: src/main/java/com/intive/patronage/toz/news/NewsController.java [hunk 1]
```diff
@@ -23,7 +24,7 @@
 @Api(description = "News operations.")
 @RestController
 @RequestMapping(value = ApiUrl.NEWS_PATH, produces = MediaType.APPLICATION_JSON_VALUE)
-class NewsController {
+public class NewsController {
```
Comment (MateuszSz93) [inline src/main/java/com/intive/patronage/toz/news/NewsController.java]: Code looks ok, but I have only one question. Why did you change it to `public`?

File: src/test/java/com/intive/patronage/toz/tokens/TokenRoleTest.java [hunk 2]
```diff
@@ -101,4 +140,48 @@ public void shouldReturnForbiddenWhenAnonymousAndNewsAchieved() throws Exception
...
+        UUID id = newsRepository.save(releasedNews).getId();
+        mockMvc.perform(get(ApiUrl.NEWS_PATH + "/" + id))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    @UseDataProvider("getCompletePet")
+    public void shouldReturnOkWhenAnonymousAndGetPetWithCompleteFieldsById(Pet pet) throws Exception {
+        UUID id = petsRepository.save(pet).getId();
+        mockMvc.perform(get(ApiUrl.PETS_PATH + "/" + id))
```
Comment (macieg-b) [inline src/test/java/com/intive/patronage/toz/tokens/TokenRoleTest.java]: From me `String.format()` is more legible

File: src/test/java/com/intive/patronage/toz/tokens/TokenRoleTest.java [hunk 3]
```diff
@@ -66,6 +78,33 @@
         return new User[]{volunteer};
     }
 
+    @DataProvider
+    public static Object[] getNewsAchieved() {
+        News news = new News();
+        news.setContents("string");
+        news.setTitle("string");
+        news.setType(News.Type.ARCHIVED);
+        return new News[]{news};
```
Comment (macieg-b) [inline src/test/java/com/intive/patronage/toz/tokens/TokenRoleTest.java]: Why don't you return normal News object?

File: src/main/java/com/intive/patronage/toz/news/NewsController.java (hunk 1 above)
Comment (Jakub-Izbicki) [inline src/main/java/com/intive/patronage/toz/news/NewsController.java]: I need to test added here roles logic in `TokenRoleTest`, which is in another package.

File: src/test/java/com/intive/patronage/toz/tokens/TokenRoleTest.java (hunk 3 above)
Comment (Jakub-Izbicki) [inline src/test/java/com/intive/patronage/toz/tokens/TokenRoleTest.java]: `@DataProvider` methods must either return `Object[][]`, `Object[]`, `String[]`, `Iterable<Iterable<?>>`, or `Iterable<?>`. It just doesn't work, when I return single `Object`, thows error during application start.

File: src/main/java/com/intive/patronage/toz/news/NewsController.java (hunk 1 above)
Comment (MateuszSz93) [inline src/main/java/com/intive/patronage/toz/news/NewsController.java]: But even when I delete this `public`, all tests are still done (10/10).

Comment (Jakub-Izbicki) [inline src/main/java/com/intive/patronage/toz/news/NewsController.java]: You are right, I used `NewsController` earlier and when I deleted it form tests I forgot to convert `NewsController.class` back to package-protected. I will make another pull request just to fix that.
